### PR TITLE
Clean up Wait() and Get() in the core worker

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

The `LocalPlasmaStoreProvider` is no longer used now that we have the `LocalMemoryStoreProvider` and we don't need to handle duplicate `ObjectID`s in the `PlasmaStoreProvider` now that they are deduplicated in the top-level object interface.

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by zhijunfu** [comment]
Thanks for doing this!  I took a quick glimpse and it looks good to me. Will take a more detailed look.

**Review by pcmoritz — COMMENTED** [review]
LGTM, these are good simplifications :)

There is some small comment.

File: src/ray/core_worker/store_provider/memory_store_provider.cc [hunk 1]
```diff
@@ -22,9 +22,20 @@ Status CoreWorkerMemoryStoreProvider::Put(const RayObject &object,
 }
 
 Status CoreWorkerMemoryStoreProvider::Get(
-    const std::vector<ObjectID> &object_ids, int64_t timeout_ms, const TaskID &task_id,
-    std::vector<std::shared_ptr<RayObject>> *results) {
-  return store_->Get(object_ids, object_ids.size(), timeout_ms, true, results);
+    std::unordered_set<ObjectID> &object_ids, int64_t timeout_ms, const TaskID &task_id,
```
[the remark this answers is not in the document]
Comment (zhijunfu) [reply src/ray/core_worker/store_provider/memory_store_provider.cc]: Yes, this looks kind of unintuitive to me too, sounds ok for an internal interface.

**Review by zhijunfu — COMMENTED** [review]
LGTM. thanks. I left a few minor comments.

File: src/ray/core_worker/store_provider/plasma_store_provider.cc [hunk 2]
```diff
@@ -79,26 +133,13 @@ Status CoreWorkerPlasmaStoreProvider::Get(
...
-      }
-    }
+    size_t previous_size = remaining.size();
+    RAY_RETURN_NOT_OK(FetchAndGetFromPlasmaStore(remaining, batch_ids, batch_timeout,
+                                                 /*fetch_only=*/false, task_id, results,
+                                                 &got_exception));
+    should_break = should_break || got_exception;
 
-    if (successes < batch_ids.size()) {
+    if ((remaining.size() - previous_size) < batch_ids.size()) {
```
Comment (zhijunfu) [inline src/ray/core_worker/store_provider/plasma_store_provider.cc]: this doesn't look right, as `previous_size` should be no less than `remaining.size()`

it should be `previous_size - remaining.size()` ?

**Review by zhijunfu — COMMENTED** [review]
LGTM. Except that there's a bug for the comparison. Thanks.

**Review by zhijunfu — APPROVED** [review]

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**